### PR TITLE
Start building the school library app

### PR DESCRIPTION
**Pull request by rbreva**
### Start building the school library app

- Create class Person
- Create class Student
- Create class Teacher

Thanks for the review.
Mauricio Mendoza

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class Person
+  attr_accessor :name, :age, :id
+
+  def initialize(age, name: 'Unkown', parent_permission: true)
+    @age = age
+    @name = name
+    @parent_permission = parent_permission
+  end
```
Comment (mikethreels) [inline person.rb]: - [ ] you added :id to the attr_accessor, this should only be added to the reader though.
- [ ] id should also be initialized as a random number.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  def of_age?(age)
+    true if age >= 18
+    false
+  end
+
+  public
+
+  def can_use_services?
+    return true if is_of_age? || @parent_permission
+  end
```
Comment (mikethreels) [inline person.rb]: - [ ] in the method `can_use_services?` your calling a method called `is_of_age?` however the actual method is called `of_age?`

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Student < Person
+  def initialize(*person, classroom)
+    super(*person)
+    @classroom = classroom
+  end
```
Comment (mikethreels) [inline student.rb]: - [ ] the constructor should include age, classroom, name and parent_permission separately similar to the example below:
```
def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
  end
```

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Teacher < Person
+  def initialize(*person, specialization)
+    super(*person)
+    @specialization = specialization
+  end
```
Comment (mikethreels) [inline teacher.rb]: - [ ] the constructor should include age, classroom, name and parent_permission separately similar to the example below:
```
def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
  end
```

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Mauricio, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name: 'Unkown', parent_permission: true)
+    @age = age
+    @name = name
+    @parent_permission = parent_permission
+    @id = rand(1...10)
+  end
```
Comment (tresorsawasawa) [inline person.rb]: - [ **OPTIONAL** ] Good :+1: job implementing the change required by the previous reviewer by assigning you id to random numbers. What do you thing if you increase its limit band value as it seems to be small? And if you want you can also use the `Random` **[class](https://ruby-doc.org/core-2.1.3/Random.html)** so your `@id` value can look to some thing like:
```ruby
@id = Random.rand(1...1000)
```

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @rbreva 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is professional :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**